chain/client/src/client_actor.rs

```diff
@@ -116,7 +117,14 @@ pub struct ClientActor {
 
     /// Synchronization measure to allow graceful shutdown.
     /// Informs the system when a ClientActor gets dropped.
-    shutdown_signal: Option<oneshot::Sender<()>>,
+    shutdown_signal: Option<broadcast::Sender<()>>,
+
+    /// `rx_config_update` is a way to get notified about the config being changed
```

Comment:
IMHO we're exposing too much details and spreading the code too much.
(my goal is - that it is clear what to do when you add a new update-able field - and you don't have to touch too many separate classes)

I'd create a class 'ConfigUpdater' - that would hold the rx_config_update, latest error etc.

And I'd wrap a code in check_triggers into the method of that class. Including also the code that you have in update_client_config.

config_updater: ConfigUpdater,


and then in check_triggers:

config_updater.update_config_if_needed(&mut self.client.config)
// or
self.client.config.update_if_needed(config_updater)